skill/__init__.py

```diff
@@ -0,0 +1,130 @@
+                        return self.Check.answer_num_range(answer_words)
+                    elif 'is_even_number' in result['answer'].keys():
+                        return self.Check.is_even_number(answer_words)
+                    elif 'is_number' in result['answer'].keys():
+                        return self.Check.is_number(answer_words)
+                    elif 'not_empty' in result['answer'].keys():
+                        return self.Check.not_empty(answer_words)
+                    elif 'check_answer' in result['answer'].keys():
+                        return self.Check.check_answer(answer_words, answer_list)
+                    # checking for existence of words in user's answer in correct answer
+                    else:
+                        answer_words = answer_words.split(' ')
+                        exist = [word for word in answer_words if word in result['answer'].keys()]
+                        # return question id from json file
+                        if len(exist) != 0:
+                            question_id = result['answer'][exist[0]]
+                            return str(question_id), ' '.join(answer_words)
+                        # ask to repeat the question and return current question id
+                        else:
+                            self.speak('Repeat, please')
+                            return str(question_id), ' '.join(answer_words)
+                else:
+                    if 'REPLACE' in result['question']:
+                    # inserting user's previous answer into the question
+                        text = self.Check.replace_question()
+                        self.speak(text)
+                        return str(question_id+1), prev_answer
+                    else:
+                        self.speak(result['question'])
+                        return str(question_id+1), prev_answer
+    
+    def execute(self):
+
+        while int(self.question_id) != 0:
+            if (len(self.answer_list) >= 3) and (self.answer_list[-3][0] == self.question_id) and (self.question_id != None):
+                print('no instructions')
+                self.speak('no instructions')
+                # self.Audio.no_instructions()
+                break
+            else:
+                print(self.question_id)
+                print(self.words_from_prev_answer)
+                result = self.conversation(self.json_list, self.question_id, self.words_from_prev_answer, self.answer_list)
+                print(result)
+                self.question_id = result[0]
+                self.words_from_prev_answer = result[1]
+                self.answer_list.append( [self.question_id, self.words_from_prev_answer])
+        else:
+            print('finish')
+            self.speak('Finished')
+    
+    def _start_instructions_prompt(self, message):
+        LOG.debug('Prompting Instructions start')
+        self.make_active()
+        start_instr = self.ask_yesno()
```

Comment:
The first arg to `ask_yesno` should be a string or vocab resource name with a prompt for the user to reply to, i.e.:

```python
self.ask_yesno("Would you like me to start the instructions?")
```

or, if you put that string in `locale/en-us/vocab/start_instructions.voc`:
```python
self.ask_yesno("start_instructions")
```